static/js/main.js

```diff
@@ -0,0 +1,168 @@
+// Due to the way processing works, some variables that are treated as constants can't be declared as such, because they must be set in `setup()`.
+// We could probably avert this issue by severing the processing library from this project, and rewriting it in terms of DOM interaction.
+
+import { timeFormatter } from "./presets.js";
+import { getTargetTime, getTime, getTimeDifference } from "./timeUtils.js";
+import TimeInformation from "./timeInformation.js";
+
+const modes = ["60x60", "25x25", "45x10", "1x1", "0.1x1", "0.02x0.02", "90x90"]; //add modes here
+
+var timerRunning = false;
+var buttons = [];
+var currentWorkTime, currentRestTime;
+var working = true; //false=resting
+var currentButtonsAmount = 0;
+var buttonWidth, buttonHeight; // Should be constant, but due to the way processing works, must be set in `setup()`.
+var endTime;
+var resetTimer = false; //for pomodoro screen
+var theText; //for pomodoro screen - global
+var timeLeft;
+var soundEffect; // Should be constant.
```

Comment:
All of these should be able to be turned into `let` without unexpected behavior. Ideally we would remove as many variables as possible.